src/ui/public/fancy_forms/kbn_form_controller.js

```diff
@@ -1,74 +1,81 @@
-import _ from 'lodash';
+export function decorateFormController($delegate, $injector) {
+  const [directive] = $delegate;
+  const FormController = directive.controller;
 
-/**
- * Extension of Angular's FormController class
- * that provides helpers for error handling/validation.
- *
- * @param {$scope} $scope
- */
-export function KbnFormController($scope, $element) {
-  const self = this;
+  class KbnFormController extends FormController {
```

Comment:
Angular 1.6.0 updated the NgModelController and FormController to use constructors and the prototype which means that I needed to update this logic to use `new directive.controller` somehow, which lead me to re-reading this code I wrote a while ago and asking why it works the way it does... The new code does exactly the same stuff as the original, tests all passed without modification, and the following improvements were made:
 - remove unnecessarily flexible decorator creation
 - use standard inheritance to extend default FormController
 - use `_` prefix for private methods rather than `$$`
 - remove unused and unnecessary `predicate` argument from invalid models getter
 -